src/simsoptpp/curveplanarfourier.cpp

```diff
@@ -0,0 +1,1557 @@
+                            - 4 * j * q_norm[1])
+                            * (- q[1] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s)
+                            +
+                            (- 4 * i * (q_norm[1] * q_norm[2] + q_norm[3] * q_norm[0]) * q_norm[2]
+                            + 2 * i * q_norm[1]
+                            + 4 * j * (q_norm[1] * q_norm[1] + q_norm[3] * q_norm[3]) * q_norm[2])
+                            * (- q[2] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s)
+                            +
+                            (- 4 * i * (q_norm[1] * q_norm[2] + q_norm[3] * q_norm[0]) * q_norm[3]
+                            + 2 * i * q_norm[0]
+                            + 4 * j * (q_norm[1] * q_norm[1] + q_norm[3] * q_norm[3]) * q_norm[3]
+                            - 4 * j * q_norm[3])
+                            * (inv_sqrt_s - q[3] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s);
+
+
+        data(m, 2, counter) = (- 4 * i * (q_norm[1] * q_norm[3] - q_norm[2] * q_norm[0]) * q_norm[0]
+                            - 2 * i * q_norm[2]
+                            - 4 * j * (q_norm[1] * q_norm[0] + q_norm[2] * q_norm[3]) * q_norm[0]
+                            + 2 * j * q_norm[1])
+                            * (- q[0] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s)
+                            +
+                            (- 4 * i * (q_norm[1] * q_norm[3] - q_norm[2] * q_norm[0]) * q_norm[1]
+                            + 2 * i * q_norm[3]
+                            - 4 * j * (q_norm[1] * q_norm[0] + q_norm[2] * q_norm[3]) * q_norm[1]
+                            + 2 * j * q_norm[0])
+                            * (- q[1] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s)
+                            +
+                            (- 2 * i * q_norm[0]
+                            - 4 * i * (q_norm[1] * q_norm[3] - q_norm[0] * q_norm[2]) * q_norm[2]
+                            - 4 * j * (q_norm[1] * q_norm[0] + q_norm[2] * q_norm[3]) * q_norm[2]
+                            + 2 * j * q_norm[3])
+                            * (- q[2] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s)
+                            +
+                            (2 * i * q_norm[1]
+                            + 4 * i * (q_norm[2] * q_norm[0] - q_norm[1] * q_norm[3]) * q_norm[3]
+                            - 4 * j * (q_norm[1] * q_norm[0] + q_norm[2] * q_norm[3]) * q_norm[3]
+                            + 2 * j * q_norm[2])
+                            * (inv_sqrt_s - q[3] * q[3] * inv_sqrt_s * inv_sqrt_s * inv_sqrt_s);
+        counter++;
+
+        for (int i = 0; i < 3; ++i) {
+            data(m, 0, counter) = 0;
+            data(m, 1, counter) = 0;
+            data(m, 2, counter) = 0;
+
+            counter++;
+        }
+    }
+}
+
+
+
+#include "xtensor-python/pyarray.hpp"     // Numpy bindings
+typedef xt::pyarray<double> Array;
+template class CurvePlanarFourier<Array>;
```

Comment:
Are these lines needed, or can they be deleted?

Reply:
I'm not sure exactly what these lines do, but they are included in both of the other curve classes, so I thought it would be safest to include them.